sbi/inference/posteriors/direct_posterior.py

```diff
@@ -583,14 +599,18 @@ def posterior_potential(
             self.device
         )
 
+        theta, log_abs_det = self.transform(theta)
+
         theta_repeated, x_repeated = DirectPosterior._match_theta_and_x_batch_shapes(
             theta, self.x
         )
 
         with torch.set_grad_enabled(track_gradients):
 
             # Evaluate on device, move back to cpu for comparison with prior.
-            posterior_log_prob = self.posterior_nn.log_prob(theta_repeated, x_repeated)
+            posterior_log_prob = (
+                self.posterior_nn.log_prob(theta_repeated, x_repeated) + log_abs_det
```

Comment:
I would also make the addition of log abs det more visible. the `posterior_log_prob` is ``untransformed", but what MCMC needs is the log prob in transformed space, no? 
So I would add it in a separate step and give it a different name.